txpool/txpool.go

```diff
@@ -352,9 +352,6 @@ func (t *TxPool) validateTx(tx *types.Transaction) error {
 		return ErrInsufficientFunds
 	}
 
-	// Make sure the transaction doesn't exceed the block limit
-	// TODO: Awaiting separate PR
-
```

Comment:
I would leave this `TODO` as it is.
AFAIK this check is there to make sure that the gas sent with the transaction doesn't exceed current block limit.
Not just to check if we can fit one regular 21k gas transaction into the block, as this new check does.